src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,169 @@
+
+            var address = QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, new Dictionary<string, string>
+            {
+                ["app_id"] = Options.ClientId,
+                ["scope"] = FormatScope(),
+                ["redirect_uri"] = redirectUri,
+                [State] = stateValue
+            });
+            return address;
+        }
+        protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync()
+        {
+            var query = Request.Query.ToDictionary(c => c.Key, c => c.Value, StringComparer.OrdinalIgnoreCase);
+            if (query.TryGetValue("auth_code", out var auth_code))
+            {
+                query["code"] = auth_code;
+                Request.QueryString = QueryString.Create(query);
+            }
+            return await base.HandleRemoteAuthenticateAsync();
+        }
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OAuthCodeExchangeContext context)
+        {
+            var sortedparams = new SortedDictionary<string, string>()
+            {
+                ["app_id"] = Options.ClientId,
+                ["method"] = "alipay.system.oauth.token",
+                ["format"] = "JSON",
+                ["charset"] = "UTF-8",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss"),
+                ["version"] = "1.0",
+                ["code"] = context.Code,
+                ["grant_type"] = "authorization_code",
+            };
+            var signparames = RSA2Sign(sortedparams);
+            if (Options.NotValidateSign)
+            {
+                signparames.Remove("timestamp");
+                signparames.Remove("sign");
+            }
+            string address = QueryHelpers.AddQueryString(Options.TokenEndpoint, signparames);
+
+            using var request = new HttpRequestMessage(HttpMethod.Get, address);
+            using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving an access token: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+            }
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+            using var alipay_system_oauth_token_response = JsonDocument.Parse(payload.RootElement.GetProperty("alipay_system_oauth_token_response").ToString());
```

Comment:
Change to `payload.RootElement.GetProperty("alipay_system_oauth_token_response").GetString()`

Reply:
In fact could just be `payload.RootElement.GetString("alipay_system_oauth_token_response")`